src/cascadia/TerminalSettingsModel/NewTabMenuEntry.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "NewTabMenuEntry.h"
+#include "JsonUtils.h"
+#include "TerminalSettingsSerializationHelpers.h"
+#include "SeparatorEntry.h"
+#include "FolderEntry.h"
+#include "ProfileEntry.h"
+#include "RemainingProfilesEntry.h"
+
+#include "NewTabMenuEntry.g.cpp"
+
+using namespace Microsoft::Terminal::Settings::Model;
+using namespace winrt::Microsoft::Terminal::Settings::Model::implementation;
+using NewTabMenuEntryType = winrt::Microsoft::Terminal::Settings::Model::NewTabMenuEntryType;
+
+static constexpr std::string_view TypeKey{ "type" };
+
+// This is a map of NewTabMenuEntryType->function<NewTabMenuEntry(Json::Value)>,
+// it allows us to choose the correct deserialization function for a given entry type
+using MenuEntryParser = std::function<winrt::com_ptr<NewTabMenuEntry>(const Json::Value&)>;
+static const std::unordered_map<NewTabMenuEntryType, MenuEntryParser> typeDeserializerMap{

```

Comment:
Yeah. `NewTabMenuEntryType` is an `enum` → Use a simple `switch` with 4 `case`s to call the functions.

Reply:
A construction like this was used elsewhere in the code (e.g. `ActionToStringMap`, `argSerializerMap` in `ActionAndArgs.cpp`) so that's why I used it but I think a switch should be fine here considering it's only used in one place.